src/main/java/javaslang/collection/Stream.java

```diff
@@ -927,10 +927,10 @@ default int indexOf(T element) {
     Stream<T> intersperse(T element);
 
     @Override
-    default int lastIndexOf(T element) {
+    default int lastIndexOf(T element, int end) {
         int result = -1, index = 0;
         for (Stream<T> stream = this; !stream.isEmpty(); stream = stream.tail(), index++) {
-            if (Objects.equals(stream.head(), element)) {
+            if (index <= end && Objects.equals(stream.head(), element)) {
```

Comment:
... then we can remove the check here and the loop ends, if `index > end`.

Would this be the same semantic?